### DISCUSSION TITLE
DNS IPv4 Rewrite (Discard AAAA) for CNAME Domains

### DISCUSSION THREAD

**Discussion by moremorefaito** [AdguardTeam/AdGuardHome#6984] [Q&A]
Hello! After reading Docs, I still cannot figure out how to:
By default, `2024.aclweb.org` CNAMEs to `acl-org.github.io`.
Lots of domains CNAME to `*.github.io`, and I want to specify `185.199.111.153` for all domains CNAME to `*.github.io`. (Due to some reaons, I have to connect via IPv4 only for this domain specially.)
I tried `||github.io^$dnstype=CNAME,dnsrewrite=185.199.111.153` in AdGuard Home Settings, finding that doesn't work.
Is there a method to specify an IP Address for domains and CNAME domains? There are hundreds of domains which CNAME to `*.github.io`, thus I could not specify them to `185.199.111.153` one by one manually.
Thanks!

**Comment by chandr1000** [comment]
I was also encountered same problem.

Assume I have domain1.lan and domain2.lan
`domain1.lan A  IN 192.168.1.10`
`www.domain1.lan CNAME  IN domain1.lan`
`www.domain2.lan CNAME  IN domain1.lan `

Then I'm gonna write rules with this syntax:
`||domain1.lan^$dnsrewrite=192.168.1.3,important`

The expected changes result should be:
```
>nslookup  www.domain2.lan

Server:  agh.lan
Address:  192.168.1.1

Non-authoritative answer:
Name:    www.domain1.lan
Addresses:  192.168.1.3
Aliases:  www.domain2.lan
```

Not this:
```
>nslookup  www.domain2.lan

Server:  agh.lan
Address:  192.168.1.1

Non-authoritative answer:
Name:    www.domain1.lan
Addresses:  192.168.1.10
Aliases:  www.domain2.lan
```

**Reply by chandr1000** [reply]
I think this should flagged as [BUG] since dnsrewrite didn't work for a domain that CNAME pointing to another domain
